### PR TITLE
Split-up of amp-bind-integrations.html for p-tag

### PR DESCRIPTION
**Pull request by sreenidhipm**
Fixes amp-bind: Speed up integration tests #8197
* Part 2: Split up the amp-bind-integrations.html test fixture into multiple files under a single new directory in test/fixtures/
* Split up for p-tag

### CODE REVIEW

**Comment by sreenidhipm** [comment]
@kmh287 Separate PR for p-tag

File: test/fixtures/amp-bind-integrations/p-tag.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <script custom-element="amp-bind" src="/dist/v0/amp-bind-0.1.max.js"></script>
+  <script custom-element="amp-iframe" src="/dist/v0/amp-iframe-0.1.max.js"></script>
+</head>
+<body>
+  <p>P TAG</p>
+  <button on="tap:AMP.setState(boundText='hello world')" id="changeTextButton"></button>
+  <button on="tap:AMP.setState(boundClass='new')" id="changeTextClassButton"></button>
+  <p id="textElement" class="original" [class]="boundClass" [text]="boundText">unbound</p>
+</body>
+</html>
```
Comment (kmh287) [inline test/fixtures/amp-bind-integrations/p-tag.html]: small nit: newline at end of file

File: test/fixtures/amp-bind-integrations/p-tag.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      background-color:green;
+    }
+    .new {
+      background-color:blue;
+    }
+  </style>
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script src="/dist/amp.js"></script>
+  <script custom-element="amp-bind" src="/dist/v0/amp-bind-0.1.max.js"></script>
+  <script custom-element="amp-iframe" src="/dist/v0/amp-iframe-0.1.max.js"></script>
```
Comment (kmh287) [inline test/fixtures/amp-bind-integrations/p-tag.html]: No need for the iframe element

File: test/fixtures/amp-bind-integrations/p-tag.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!doctype html>
+<html ⚡>
+<head>
+  <meta charset="utf-8">
+  <title>amp-bind integration test</title>
```
Comment (kmh287) [inline test/fixtures/amp-bind-integrations/p-tag.html]: nit: change title to match what this test fixture is for

Comment (sreenidhipm) [inline test/fixtures/amp-bind-integrations/p-tag.html]: Done!

File: test/fixtures/amp-bind-integrations/p-tag.html (hunk 2 above)
Comment (sreenidhipm) [inline test/fixtures/amp-bind-integrations/p-tag.html]: Done!

File: test/fixtures/amp-bind-integrations/p-tag.html (hunk 1 above)
Comment (sreenidhipm) [inline test/fixtures/amp-bind-integrations/p-tag.html]: Done!

**Comment by sreenidhipm** [comment]
Do you want me to create a separate test file like test-bind-integration.js and have the tests only for the p-tag?

**Comment by kmh287** [comment]
Yes exactly. :+1: 

Feel free to ping me on slack (kmh287) with questions.

**Comment by sreenidhipm** [comment]
Done!

File: extensions/amp-bind/0.1/test/test-bind-integration-ptag.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+ /** @return {!Promise} */
+  function waitForBindApplication() {
+   // Bind should be available, but need to wait for actions to resolve
+   // service promise for bind and call setState.
+    return bindForDoc(ampdoc).then(unusedBind =>
+       waitForEvent('amp:bind:setState'));
+  }
+
+  describe('text integration', () => {
```
Comment (kmh287) [inline extensions/amp-bind/0.1/test/test-bind-integration-ptag.js]: Let's remove this describe block and put the tests directly under the top-level describe.

File: extensions/amp-bind/0.1/test/test-bind-integration-ptag.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createFixtureIframe} from '../../../../testing/iframe';
+import {bindForDoc} from '../../../../src/bind';
+import {ampdocServiceFor} from '../../../../src/ampdoc';
+
+describe.configure().retryOnSaucelabs().run('amp-bind', function() {
```
Comment (kmh287) [inline extensions/amp-bind/0.1/test/test-bind-integration-ptag.js]: 'amp-bind' -> 'amp-bind text integration'

File: test/fixtures/amp-bind-integrations/p-tag.html (hunk 1 above)
Comment (kmh287) [inline test/fixtures/amp-bind-integrations/p-tag.html]: I'm not seeing it on my end.

File: extensions/amp-bind/0.1/test/test-bind-integration-ptag.js (hunk 4 above)
Comment (sreenidhipm) [inline extensions/amp-bind/0.1/test/test-bind-integration-ptag.js]: done

File: extensions/amp-bind/0.1/test/test-bind-integration-ptag.js (hunk 5 above)
Comment (sreenidhipm) [inline extensions/amp-bind/0.1/test/test-bind-integration-ptag.js]: done

File: extensions/amp-bind/0.1/test/test-bind-integration-ptag.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createFixtureIframe} from '../../../../testing/iframe';
+import {bindForDoc} from '../../../../src/bind';
```
Comment (kmh287) [inline extensions/amp-bind/0.1/test/test-bind-integration-ptag.js]: I think if you change the import location to `'../../../../src/services`the continuous integration will pass and you can get this PR in :+1:

Comment (sreenidhipm) [inline extensions/amp-bind/0.1/test/test-bind-integration-ptag.js]: Done! Sorry for the delay in updating.

**Pull request merged**